File: 001-BitwiseOperators/009.powerOfTwo.js
// Naive Method
const powerOfTwoN = (n) => {
  if (n == 0) {
    return false;
  }
  while (n != 1) {
    if (n % 2 != 0) {
      return false;
    }
    n = parseInt(n / 2);
  }
  return true;
};

// Efficient, The number must have only one set bit (n & n-1) sets rightmost bit 0

const powerOfTwo = (n) => {
  if (n == 0) {
    return false;
  }
  return (n & (n - 1)) === 0;
};

File: GeeksForGeeks/DS-Course/001-BitwiseOperators/009.powerOfTwo.py
# Naive Method
def powerOfTwoN(n):
    if n == 0:
        return False
    
    while n != 1:
        if n % 2 != 0:
            return False
        n = n // 2
    return True

# print(powerOfTwoN(514))


# Efficient, The number must have only one set bit

def powerOfTwo(n):
    if n == 0:
        return False
    return (n & (n-1)) == 0

print(powerOfTwo(40))